Src/BootCamp.Chapter/CredentialsManager.cs

```diff
@@ -0,0 +1,168 @@
+using System.IO;
+
+namespace BootCamp.Chapter
+{
+    internal class CredentialsManager
+    {
+        private const string CredentialsPath = @"..\..\..\Database\Credentials.txt";
+
+        public CredentialsManager()
+        {
+            if (!File.Exists(CredentialsPath))
+            {
+                File.WriteAllText(CredentialsPath, "");
+            }
+        }
+
+        public void Register()
+        {
+            if (RegisterUser())
+            {
+                Console.WriteLine("You have successfully registered.\n");
+            }
+            else
+            {
+                Console.WriteLine("Sorry, your registration was unsuccessful.  The username is already taken.\n");
+            }
+        }
+
+        private bool RegisterUser()
+        {
+            const string CarriageReturn = "\r\n";
+            const string Delimiter = ",";
+
+            var user = GetNameAndPassword();
+
+            if (IsNameRegistered(user.Username))
+            {
+                return false;
+            }
+
+            var encodedPassword = new Cryptography().Encode(user.Password);
+            File.AppendAllText(CredentialsPath, user.Username);
+            File.AppendAllText(CredentialsPath, Delimiter);
+            File.AppendAllText(CredentialsPath, encodedPassword);
+            File.AppendAllText(CredentialsPath, CarriageReturn);
+
+            return true;
+        }
+
+        public bool Login()
+        {
+            var user = GetNameAndPassword();
+
+            if (AuthenticateUser(user))
+            {
+                Console.WriteLine("You have sucessfuly logged in.\n");
+                return true;
+            }
+
+            Console.WriteLine("Sorry, we don't recognize you.\n");
+            return false;
+        }
+
+        private Credentials GetNameAndPassword()
+        {
+            string username = "";
+
+            do
+            {
+                username = Utilities.PromptText("Enter username:");
+            } while (string.IsNullOrEmpty(username.Trim()));
+
+            string password = "";
+
+            do
+            {
+                password = Utilities.PromptPassword("Enter password:");
+            } while (string.IsNullOrEmpty(password.Trim()));
+
+            return new Credentials(username, password);
+        }
+
+
+
+        private bool IsNameRegistered(string username)
+        {
+            var user = GetUser(username);
+
+            if (user.Username == null)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private bool AuthenticateUser(Credentials user)
+        {
+            var storedUser = GetUser(user.Username);
+
+            if (storedUser.Username == null || storedUser.Password != new Cryptography().Encode(user.Password))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private Credentials GetUser(string username)
```

Comment:
You get Credentials. Maybe user credentials. But not user itself.

Reply:
Changes made when the code was coded saying getting user information, such as "GetUser" name.  When I switched to using the term "Credential", I did not change the all the places in the code where was still referring as "User".  Thank you for catching this with your sharp eyes.  It showed me that I have to be more careful when making global name changes.